pkg/orchestration/wiring/postgres/plugin.go

```diff
@@ -123,65 +92,57 @@ func shapes(resource *orch_v1.StateResource, smithResource *smith_v1.Resource, c
 		"READPASSWORD": "data.readpassword",
 		"READURL":      "data.readurl",
 	}
-
-	// If the postgres has an RDS dependency with read replica enabled, it
-	// produces some additional environment variables for the read replica
-	var foundSharedDb []*knownshapes.SharedDb
-	for _, dep := range context.Dependencies {
-		sharedDb, found, err := knownshapes.FindSharedDbShape(dep.Contract.Shapes)
+	if found {
+		var sharedDbShape *knownshapes.SharedDb
+		sharedDbShape, found, err = knownshapes.FindSharedDbShape(sharedDbDep.Contract.Shapes)
```

Comment:
This functions now has a single place where we are looking for the shape we need.